cmd/service_linux.go

```diff
@@ -0,0 +1,316 @@
+
+		// assumes dir is in cloudfuse repo dir
+		serviceData, err := collectServiceData(fmt.Sprintf("%s/setup/cloudfuse.service", dir))
+		if err != nil {
+			return fmt.Errorf("error collecting data from cloudfuse.service file due to the following error: [%s]", err)
+		}
+		serviceUser := serviceData["User"]
+		setUser(serviceUser)
+
+		// 4. run systemctl daemon-reload
+		systemctlDaemonReloadCmd := exec.Command("sudo", "systemctl", "daemon-reload")
+		err = systemctlDaemonReloadCmd.Run()
+		if err != nil {
+			return fmt.Errorf("failed to run 'systemctl daemon-reload' command due to following error: [%s]", err.Error())
+		}
+
+		// 5. Enable the service to start at system boot
+		systemctlEnableCmd := exec.Command("sudo", "systemctl", "enable", serviceFile)
+		err = systemctlEnableCmd.Run()
+		if err != nil {
+			return fmt.Errorf("failed to run 'systemctl daemon-reload' command due to following error: [%s]", err.Error())
+		}
+		return nil
+	},
+}
+
+var uninstallCmd = &cobra.Command{
+	Use:               "uninstall",
+	Short:             "Uninstall the startup process for Cloudfuse. Requires running as admin.",
+	Long:              "Uninstall the startup process for Cloudfuse. Requires running as admin.",
+	SuggestFor:        []string{"uninst", "uninstal"},
+	Example:           "cloudfuse service uninstall",
+	FlagErrorHandling: cobra.ExitOnError,
+	RunE: func(cmd *cobra.Command, args []string) error {
+
+		// 1. find and remove cloudfuse.service from /etc/systemd/system and run systemctl daemon-reload
+		removeFileCmd := exec.Command("sudo", "rm", "/etc/systemd/system/cloudfuse.service")
+		err := removeFileCmd.Run()
+		if err != nil {
+			return fmt.Errorf("failed to delete cloudfuse.service file from /etc/systemd/system due to following error: [%s]", err.Error())
+		}
+
+		systemctlDaemonReloadCmd := exec.Command("sudo", "systemctl", "daemon-reload")
+		err = systemctlDaemonReloadCmd.Run()
+		if err != nil {
+			return fmt.Errorf("failed to run 'systemctl daemon-reload' command due to following error: [%s]", err.Error())
+		}
+
+		return nil
+	},
+}
+
+//--------------- command section ends
+
+func collectServiceData(serviceFilePath string) (map[string]string, error) {
+	serviceFile, err := os.Open("./setup/cloudfuse.service")
+
+	if err != nil {
+		fmt.Println("Error opening file:", err)
+		return nil, err
+	}
+
+	defer serviceFile.Close()
+
+	scanner := bufio.NewScanner(serviceFile)
+	serviceData := make(map[string]string)
+	for scanner.Scan() {
+		line := scanner.Text()
+		if strings.Contains(line, "User=") {
+			parts := strings.SplitN(line, "=", 2)
+			key := strings.TrimSpace(parts[0])
+			value := strings.TrimSpace(parts[1])
+			serviceData[key] = value
+		}
+	}
+	if err := scanner.Err(); err != nil {
+		fmt.Println("Error reading file:", err)
+		return nil, err
+	}
+	return serviceData, nil
+}
+
+func newSericeFile(mountPath string, configPath string, dir string) (string, error) {
```

Comment:
In this function you assume that configPath is just located in the current directory dir. Could you instead get the full path to the config file, since the user needs to pass the location of configPath anyway.